src/dotnet/APIView/APIViewWeb/BlobCommentRepository.cs

```diff
@@ -23,53 +23,129 @@ public BlobCommentRepository(IConfiguration configuration)
 
         private BlobContainerClient ContainerClient { get; }
 
-        public async Task<CommentModel> ReadCommentContentAsync(string id)
+        /// <summary>
+        /// Return the blobs contained in the comments blob container.
+        /// </summary>
+        /// <returns>A collection of the blobs in the container.</returns>
+        public async Task<List<BlobItem>> FetchBlobsAsync()
         {
-            var result = await ContainerClient.GetBlockBlobClient(id).DownloadAsync();
+            var segment = ContainerClient.GetBlobsAsync(new GetBlobsOptions() { IncludeMetadata = true });
 
-            using (StreamReader reader = new StreamReader(result.Value.Content))
+            var blobs = new List<BlobItem>();
+            await foreach (var item in segment)
             {
-                CommentModel comment = JsonSerializer.Parse<CommentModel>(reader.ReadToEnd());
-                return comment;
+                blobs.Add(item);
             }
+            return blobs;
         }
 
-        public async Task<CommentModel[]> FetchCommentsAsync(string assemblyID)
+        /// <summary>
+        /// Return all comments written for review of the assembly with the provided ID.
+        /// </summary>
+        /// <param name="assemblyId">The ID of the assembly to have its comments read.</param>
+        /// <returns>The comments existing for the specified assembly if it exists, or null if no assembly has the specified ID.</returns>
+        public async Task<AssemblyCommentsModel> FetchCommentsAsync(string assemblyId)
         {
-            var segment = await ContainerClient.ListBlobsFlatSegmentAsync(options: new BlobsSegmentOptions() { Details = new BlobListingDetails() { Metadata = true } });
+            var blobs = await FetchBlobsAsync();
 
-            var comments = new List<CommentModel>();
-            foreach (var item in segment.Value.BlobItems.OrderBy(blob => blob.Properties.CreationTime))
+            foreach (var blob in blobs.OrderBy(blob => blob.Properties.CreationTime))
```

Comment:
You are still doing an inefficient thing of enumerating all blobs to find the right one.

Reply:
Yeah, reuse the GUID